src/ranged.cpp

```diff
@@ -2384,7 +2384,7 @@ void target_ui::update_target_list( player &pc )
     // Get targets in range and sort them by distance (targets[0] is the closest)
     // FIXME: get_targetable_creatures does not consider some of the visible creatures
     //        as targets (e.g. those behind fences), but you can still see and shoot them
-    targets = pc.get_targetable_creatures( range );
+    targets = pc.get_targetable_creatures( range, false );
```

Comment:
This `false` should be `mode == TargetMode::Reach`. Otherwise pressing `TAB` while selecting a target for melee attack would consider the ones behind glass as valid.